File: Day2/islandPerimeter.cpp
class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        if(grid.size() == 0)
            return 0;
        
        int perimeter = 0;
        int m = grid.size(), n = grid[0].size();
        
        for(int i = 0; i < m; i++){
            for(int j = 0; j < n; j++){
                if(grid[i][j] == 1){
                    if((i - 1 < 0) || (i - 1 >= 0 && grid[i-1][j] == 0))
                        perimeter++;
                    if((i + 1 == m) || (i + 1 < m && grid[i+1][j] == 0))
                        perimeter++;
                    if((j - 1 < 0) || (j - 1 >= 0 && grid[i][j-1] == 0))
                        perimeter++;
                    if((j + 1 == n) || (j + 1 < n && grid[i][j+1] == 0))
                        perimeter++;
                }
            }
        }
        
        return perimeter;
    }
};

File: Day2/islandPerimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        c=0
        for i in range(len(grid)) :
            for j in range(len(grid[i])) :
                if grid[i][j]==1 :
                    if i==0 :
                        c+=1
                    else :
                        if grid[i-1][j]!=1 :
                            c+=1
                    if i==len(grid)-1 :
                        c+=1
                    else :
                        if grid[i+1][j]!=1 :
                            c+=1
                    if j==0 :
                        c+=1
                    else :
                        if grid[i][j-1]!=1 :
                            c+=1
                    if j==len(grid[i])-1 :
                        c+=1
                    else :
                        if grid[i][j+1]!=1 :
                            c+=1
        return c
            
